### PR TITLE
Fix creating optional file

### LINKED ISSUE #546: Optional path not working
Hi.
Following project spec should make optional file reference, but it's not made.

```yml
name: MyProject
targets:
  MyFramework:
    platform: iOS
    type: framework
    sources:
      - path: Sources
      - path: Sources/File.swift
        optional: true
```

#527 might be related.

xcodegen 2.3.0

Thanks.

### PR DESCRIPTION
**Pull request by KyoheiG3**
Fixes #546.

### CODE REVIEW

**Comment by yonaskolb** [comment]
Hi @KyoheiG3. I'm not sure what this fixes?

File: Sources/XcodeGenKit/SourceGenerator.swift [hunk 1]
```diff
@@ -521,6 +516,11 @@ class SourceGenerator {
             sourceFiles.append(sourceFile)
 
         case .group:
+            if targetSource.optional {
+                // Is optional so if it doesn't exist just return an empty array
+                return []
```
Comment (yonaskolb) [inline Sources/XcodeGenKit/SourceGenerator.swift]: This isn't checking if a directory exists, so it would always return nothing

**Comment by ra1028** [comment]
Hi @yonaskolb 

This pull request is my colleague's.
We want to fix to reference auto-generated files that hasn't been generated yet when runs xcodegen.
For example, case of generate localized strings file using SwiftGen in build phases.
Requires to generate a Xcode project before runs SwiftGen in the pre-build phase.
So we expect like following: 
Screen Shot 2019-03-28 at 23 38 47

How should we fix?

**Comment by AlexisQapa** [comment]
I think there is a pre gen hook or at least it was mentioned in another issue. Right now I deal with this by either having blank files (we don't use swiftgen for storyboard so files are known in advance) or running swiftgen once before Xcode gen.

**Comment by yonaskolb** [comment]
Comments left in https://github.com/yonaskolb/XcodeGen/issues/546
@AlexisQapa suggestions should resolve this issue.
Closing this for now

**Pull request closed without merging**